### PR TITLE
Feature: Fully Qualified App Name

### PR DESCRIPTION
**Pull request by michael-valdron**
**Please specify the area for this PR**

operator

**What does does this PR do / why we need it**:

Adds the fully qualified app name found under the [helm chart](https://github.com/devfile/registry-support/blob/b950a067ef2eb2a1bb08ea1658bedeb1b600e722/deploy/chart/devfile-registry/templates/_template.tpl#L42) and refactors to match usage. Additionally adds truncation to both app name and fully qualified app name as seen in the helm chart.

**Which issue(s) this PR fixes**:

Fixes #?

fixes devfile/api#1441, devfile/api#1442

**PR acceptance criteria**:

- [x] Test Coverage 
    - Are your changes sufficiently tested, and are any applicable test cases added or updated to cover your changes?
- [x] Gosec scans 
  - Fix all MEDIUM and higher findings and/or annotate a finding per gosec instructions: https://github.com/securego/gosec#annotating-code to address why a finding is not a security issue

Documentation
- [ ] Does the registry operator documentation need to be updated with your changes?

**How to test changes / Special notes to the reviewer**:

A reviewer will need to run these commands to run the integration tests locally or else the `make test-integration` will likely fail:

1. Run `make install-cert`
2. Wait for cert-manager to be ready, its under the `cert-manager` namespace
3. Run `make install && make deploy`
4. Wait for registry-operator to be ready, its under the `registry-operator-system` namespace
5. Run `make test-integration`

### CODE REVIEW

**Comment by michael-valdron** [comment]
Review status is waiting to resolve an issue discovered during testing: https://github.com/devfile/api/issues/1441#issuecomment-2050188151

**Comment by michael-valdron** [comment]
> Does the registry operator documentation need to be updated with your changes?

Out of issue scope.

**Comment by Jdubrick** [comment]
Unit tests are passing for me - Waiting on a cluster to spin up so I can run the integration tests because they don't want to work on my local crc

**Comment by michael-valdron** [comment]
> Unit tests are passing for me - Waiting on a cluster to spin up so I can run the integration tests because they don't want to work on my local crc

@Jdubrick Updated the reviewer notes on the PR description with the steps to ensure `make test-integration` runs properly.

**Comment by Jdubrick** [comment]
/lgtm

All unit and integration tests are passing for me. I'm okay with it being merged but if @thepetk wants to take a look before that's okay with me.

File: pkg/registry/constants.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+const maxTruncLength = 63
```
Comment (thepetk) [inline pkg/registry/constants.go:19]: Out of curiosity, what is the difference between the `pkg/registry/contstants.go` & `pkg/registry/defaults.go`?

**Review by thepetk — COMMENTED** [review]
Overall looks good to me, left only a minor question

File: pkg/registry/constants.go (hunk 1 above)
Comment (michael-valdron) [reply pkg/registry/constants.go:19]: @thepetk Yes, `defaults.go` is everything related to default values in the `DevfileRegistry` CR whereas `constants.go` is just global constants such as the one we have for index server: https://github.com/devfile/registry-support/blob/b950a067ef2eb2a1bb08ea1658bedeb1b600e722/index/server/pkg/server/constants.go

**Review by thepetk — APPROVED** [review]
/lgtm

**Pull request merged**